apps/bug-cd/src/app/main-navigation.component.ts

```diff
@@ -43,20 +44,34 @@ export class NavigationComponent {
   template: `
     <ng-container *ngIf="info$ | async as info">
       <ng-container *ngIf="info !== null; else noInfo">
-        <app-nav [menus]="getMenu(info)" />
+        <!-- Switching to a variable, instead of calling a function everytime a CD hits -->
+        <app-nav [menus]="menus" />
       </ng-container>
     </ng-container>
 
     <ng-template #noInfo>
-      <app-nav [menus]="getMenu('')" />
+      <!-- Switching to a variable, instead of calling a function everytime a CD hits -->
+      <app-nav [menus]="menus" />
     </ng-template>
   `,
   host: {},
 })
 export class MainNavigationComponent {
   private fakeBackend = inject(FakeServiceService);
-
-  readonly info$ = this.fakeBackend.getInfoFromBackend();
+  info$: any;
```

Comment:
Don't forget to use types 😃 

Should avoid using `any` in most cases if possible. Can use `unknown` if you don't actually know the type, but in this case, we do.

Reply:
as @AlDrag mentionned, always type your properties. avoid `any` or use it when you really have no other choice.

`unknown` is not a good type here as well. And `unknow` type are not easy to deal with.

Reply:
Absolutely! Resolved, thanks!